ambry-api/src/main/java/com/github/ambry/quota/QuotaChargeCallback.java

```diff
@@ -51,22 +50,23 @@ public void charge(long chunkSize) throws RouterException {
           if (throttlingRecommendation != null && throttlingRecommendation.shouldThrottle() && shouldThrottle) {
```

Comment:
I keep forgetting what is the purpose of passing another `shouldThrottle` parameter to the callback, why do we need it?

Reply:
in that case, might as well just don't pass the quota callback.

Reply:
I don't follow whats the argument here? This PR is basically for introducing a uniform QuotaException to use for quota use cases instead of using RestServiceException or RouterException. The above code is not being changed, and will be refactored out in future PRs.